app/jobs/scheduled/process_shelved_notifications.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ProcessShelvedNotifications < ::Jobs::Scheduled
+    every 5.minutes
+
+    def execute(args)
+      sql = <<~SQL
+        SELECT n.id FROM notifications AS n
+        INNER JOIN do_not_disturb_timings AS dndt ON n.user_id = dndt.user_id
+        WHERE n.processed = false
+        AND dndt.ends_at <= :now
+      SQL
+
+      now = Time.zone.now
+      notification_ids = DB.query_single(sql, now: now)
+
+      Notification.where(id: notification_ids).each do |notification|
```

Comment:
What if there's an error when processing notifications? Should we use a transaction here or rescue errors and continue?